### PR TITLE
Release Drafter: Automatically replace GitHub handles with names and links

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Closes https://github.com/GenericMappingTools/pygmt/issues/2775.

The `replacers` almost works, but it has some trouble with newlines `\n`.

In my own fork, the draft release (https://github.com/seisman/pygmt/releases, but it's likely you can't see a draft release) looks like this:
image

### CODE REVIEW

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -27,6 +27,19 @@ exclude-contributors:
 category-template: '### $TITLE'
 change-template: '* $TITLE ([#$NUMBER]($URL))'
 sort-by: 'title'
+replacers:
+  - search: '/@(\w+)(,\s| and )?/g'
+    replace: '- [@$1](https://github.com/$1)\\n'
```
Comment (yvonnefroehlich) [inline .github/release-drafter.yml:32]: This is probably more for my understanding: Why are there now two backslashes (`\\`) before the `n`? In comment https://github.com/GenericMappingTools/pygmt/issues/2775#issuecomment-1784454161 it was only one backslash (`\`).

Comment (seisman) [reply .github/release-drafter.yml]: I first tried `\n`, then found it didn't work, then tried `\\n`, still no luck.

File: .github/release-drafter.yml [hunk 2]
```diff
@@ -27,6 +27,19 @@ exclude-contributors:
 category-template: '### $TITLE'
 change-template: '* $TITLE ([#$NUMBER]($URL))'
 sort-by: 'title'
+replacers:
+  - search: '/@(\w+)(,\s| and )?/g'
+    replace: '- [@$1](https://github.com/$1)\n'
```
Comment (seisman) [inline .github/release-drafter.yml:32]:
```suggestion
  - search: '/@(\w+)?/g'
    replace: '[@$1](https://github.com/$1)'
```

**Comment by seisman** [comment]
```
  - search: '/@(\w+)(,\s| and )?/g'
    replace: '- [@$1](https://github.com/$1)\n'
```
The above regex produces the following output:

image

It's likely because Markdown doesn't support `\n`. There may be some workarounds but I don't think it is worth my time. 

In commit 4f8236ad8656553672b27b29cf2ae5ed07d323e4, I've changed the regex to 
```
  - search: '/@(\w+)?/g'
    replace: '[@$1](https://github.com/$1)'
```
The new regex produces:
image

It looks good to me. When making a new release, we just need to convert it to a bulleted list and sort the names based on number of commits.

**Review by michaelgrund — APPROVED** [review]

File: .github/release-drafter.yml [hunk 3]
```diff
@@ -27,6 +27,19 @@ exclude-contributors:
...
+  - search: '@maxrjones'
+    replace: 'Max Jones'
+  - search: '@michaelgrund'
+    replace: 'Michael Grund'
+  - search: '@seisman'
+    replace: 'Dongdong Tian'
+  - search: '@weiji14'
+    replace: 'Wei Ji'
+  - search: '@yvonnefroehlich'
+    replace: 'Yvonne Fröhlich'
```
Comment (yvonnefroehlich) [inline .github/release-drafter.yml:42]: Maybe we should include @willschlitzer in this list?

```suggestion
  - search: '@weiji14'
    replace: 'Wei Ji'
  - search: '@willschlitzer'
    replace: 'Will Schlitzer'
  - search: '@yvonnefroehlich'
    replace: 'Yvonne Fröhlich'
```

File: .github/release-drafter.yml [hunk 4]
```diff
@@ -27,6 +27,19 @@ exclude-contributors:
...
+  - search: '/@(\w+)?/g'
+    replace: '[@$1](https://github.com/$1)'
+  - search: '@maxrjones'
+    replace: 'Max Jones'
+  - search: '@michaelgrund'
+    replace: 'Michael Grund'
+  - search: '@seisman'
+    replace: 'Dongdong Tian'
+  - search: '@weiji14'
+    replace: 'Wei Ji'
```
Comment (yvonnefroehlich) [inline .github/release-drafter.yml:40]:
```suggestion
  - search: '@weiji14'
    replace: 'Wei Ji Leong'
```

File: .github/release-drafter.yml (hunk 3 above)
Comment (seisman) [reply .github/release-drafter.yml:42]: Of course. Thanks for catching it.

**Review by weiji14 — APPROVED** [review]

File: .github/release-drafter.yml [hunk 5]
```diff
@@ -27,6 +27,21 @@ exclude-contributors:
...
 sort-by: 'title'
+replacers:
+  - search: '/@(\w+)?/g'
+    replace: '[@$1](https://github.com/$1)'
+  - search: '@maxrjones'
+    replace: 'Max Jones'
+  - search: '@michaelgrund'
+    replace: 'Michael Grund'
+  - search: '@seisman'
+    replace: 'Dongdong Tian'
```
[the remark this answers is not in the document]
Comment (seisman) [reply .github/release-drafter.yml:38]: They're already sorted alphabetically by GitHub handles, no?

[the remark this answers is not in the document]
Comment (weiji14) [reply .github/release-drafter.yml:38]: Oh right, was looking at full names, nvm.

**Pull request merged**